### PR TITLE
fix: add doctor diagnostic tests for v2 crypto

### PR DESCRIPTION
**Pull request by shetzel**
### What does this PR do?
REQUIRES: https://github.com/forcedotcom/sfdx-core/pull/810

Adds doctor diagnostic tests for v2 crypto.

### What issues does this PR fix or reference?
@W-14999859@

### CODE REVIEW

File: messages/diagnostics.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# sfCryptoV2Support
+
+This CLI does not yet support v2 crypto. All plugins and libraries must use at least version 6.7.0 of `@salesforce/core` to support v2 crypto. This will not cause a problem with CLI authentication unless a v2 crypto key has been generated.
```
Comment (jshackell-sfdc) [inline messages/diagnostics.md:3]:
```suggestion
Your current installation of Salesforce CLI, including all the plugins you've linked and installed, doesn't yet support v2 crypto. All plugins and libraries must use at least version 6.7.0 of `@salesforce/core` to support v2 crypto. You're generally still able to successfully authenticate with your current CLI installation, but not if you generate a v2 crypto key.
```

File: messages/diagnostics.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+# sfCryptoV2Unstable
+
+This CLI is using v2 crypto without proper library support. This can cause authentication failures. Switching back to v1 crypto is recommended.
```
Comment (jshackell-sfdc) [inline messages/diagnostics.md:7]:
```suggestion
Your current installation of Salesforce CLI, including all the plugins you've linked and installed, is using v2 crypto without proper library support, which can cause authentication failures. We recommend that you switch back to v1 crypto.
```

File: messages/diagnostics.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# sfCryptoV2Desired
+
+SF_CRYPTO_V2=true is set but v1 crypto is actually in use. This CLI supports using v2 crypto. If this is desired please follow the instructions in the docs (provide link).
```
Comment (jshackell-sfdc) [inline messages/diagnostics.md:11]:
```suggestion
SF_CRYPTO_V2=true is set but v1 crypto is actually in use. Your Salesforce CLI installation supports using v2 crypto. If you desire this behavior, follow the instructions in the documentation (provide link).
```

**Review by jshackell-sfdc — CHANGES_REQUESTED** [review]

**Review by cristiand391 — APPROVED** [review]

**Comment by cristiand391** [comment]
**QA notes**:

with sf v2.33.3

### v2 crypto + generic unix keychain

✅ `CLI supports v2` warn, `CLI using stable v2 crypto` fails
two 3rd party plugins use old sfdx-core, suggestions mention to go back to v1 crypto

✅ after uninstalling all 3rd party plugins, both v2 crypto diagnostics pass

✅ diagnostics also detect linked plugins with old sfdx-core

### v1 crypto + generic unix keychain

✅ no plugins, diagnostics detect full v2 crypto support and that v1 is being used

### v1 crypto + macos keychain

✅ detect v2 crypto supported, can't tell which one is being used.


### CLI installed via installer, disabled installed node/nvm (`nvm unload`)

install a 3rd party plugin so that the diagnostic tries to run `npm`, also make sure `npm` is not available globally (`which npm`)

❌  diagnostic fails with `npm: command not found`:
Screenshot 2024-03-19 at 16 44 16

we can't rely on `npm` being installed, plugin-trust/plugins bundle npm then use that to run commands:

plugin-trust:
https://github.com/salesforcecli/plugin-trust/blob/fd3540ff98d4749c8d06a614114351b4e176a478/src/shared/npmCommand.ts#L82

plugin-plugins:
https://github.com/oclif/plugin-plugins/blob/5336f76963892f2e6dc7f4a3d7fa5787dcd02229/src/util.ts#L93

if it's not worth maybe we can disable checks that need to run npm if it's not available.

File: messages/diagnostics.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+Your current installation of Salesforce CLI, including all the plugins you've linked and installed, doesn't yet support v2 crypto. All plugins and libraries must use at least version 6.7.0 of `@salesforce/core` to support v2 crypto. You're generally still able to successfully authenticate with your current CLI installation, but not if you generate a v2 crypto key.
+
+# sfCryptoV2Unstable
+
+Your current installation of Salesforce CLI, including all the plugins you've linked and installed, is using v2 crypto without proper library support, which can cause authentication failures. We recommend that you switch back to v1 crypto.
+
+# sfCryptoV2Desired
+
+SF_CRYPTO_V2=true is set but v1 crypto is actually in use. Your Salesforce CLI installation supports using v2 crypto. If you desire this behavior, follow the instructions in the documentation (provide link).
```
Comment (jshackell-sfdc) [inline messages/diagnostics.md:11]:
```suggestion
SF_CRYPTO_V2=true is set in your environment, but you're actually using v1 crypto. Your Salesforce CLI installation supports using v2 crypto. If you desire this behavior, follow the instructions in the documentation (provide link).
```

**Review by jshackell-sfdc — APPROVED** [review]

**Review by cristiand391 — APPROVED** [review]

**Comment by cristiand391** [comment]
**QA update**:

✅ can see auth crypto diagnostics when `npm` is available on `PATH`
✅ if no `npm`, test prints:
```
  * Couldn't determine latest CLI version due to:
/bin/sh: npm: command not found
```

with exit code 0

**Pull request merged**